src/codemagic/cli/argument/common_argument_types.py

```diff
@@ -7,6 +7,13 @@
 
 class CommonArgumentTypes:
 
+    @staticmethod
+    def maybe_dir(path_str: str) -> pathlib.Path:
+        path = pathlib.Path(path_str).expanduser()
+        if path.is_dir() or not path.exists():
```

Comment:
nip

In case the directory structure is

```
dir/
  file
```

and the path is `dir/file/dir`, it will fail to create directories with something like

```
mkdir: file: File exists
```

Reply:
Indeed. Good catch.

Reply:
Addressed in 6c9f3bf.